pkg/util/hlc/hlc.go

```diff
@@ -125,6 +134,61 @@ func NewClock(physicalClock func() int64, maxOffset time.Duration) *Clock {
 	}
 }
 
+// StartMonitoringForwardClockJumps starts a goroutine to update the clock's
+// maxForwardClockJump based on the values pushed in maxForwardClockJumpCh.
+//
+// This also keeps lastPhysicalTime up to date to avoid spurious jump errors.
+// maxForwardClockJumpCh is a channel from which the tolerated physical clock
+// forward jump threshold is read.
+//
+// A nil channel or a value of 0 for maxForwardClockJump disables checking clock
+// jumps between two successive reads of the physical clock.
+//
+// This should only be called once per clock, and will return an error if called
+// more than once
+//
+// tickerFn is used to create a new ticker
+//
+// tickCallback is called whenever maxForwardClockJumpCh or a ticker tick is
+// processed
+func (c *Clock) StartMonitoringForwardClockJumps(
+	maxForwardClockJumpCh <-chan time.Duration,
+	tickerFn func(d time.Duration) *time.Ticker,
+	tickCallback func(),
+) error {
+	alreadyMonitoring := c.setMonitoringClockJump()
+	if alreadyMonitoring {
+		return errors.New("clock jumps are already being monitored")
+	}
+
+	go func() {
+		ticker := tickerFn(c.maxOffset)
+		ticker.Stop()
```

Comment:
What's the purpose of this? Seems like the ticker won't tick if you stop here. Am I missing something?

Reply:
This ticker is being started later based on whether clock jump check is enabled.